From fa215a9d2115b72e71694a975f145050d0bd1f8f Mon Sep 17 00:00:00 2001
From: Pankaj Bharadiya <pankaj.laxminarayan.bharadiya@intel.com>
Date: Mon, 10 Jul 2017 12:02:29 +0530
Subject: [PATCH 4216/4301] ASoC:CNL:cnl_svfpga:Get codec from snd_soc_codec
 struct
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

commit afa2ecf81c1177e255646f9040a5bddc4a7e817f from
git://git@github.com:01org/linux-intel-4.9.git

codec list is part of snd_soc_codec.
This fixes below build error.
error: ‘struct snd_soc_component’ has no member named ‘card_list’

Change-Id: I69e2484082587f4d30a4b5692d282e76fba6de25
Signed-off-by: Pankaj Bharadiya <pankaj.laxminarayan.bharadiya@intel.com>
---
 sound/soc/intel/boards/cnl_svfpga.c |   10 +++++-----
 1 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/sound/soc/intel/boards/cnl_svfpga.c b/sound/soc/intel/boards/cnl_svfpga.c
index 3d6309a..80f258a 100644
--- a/sound/soc/intel/boards/cnl_svfpga.c
+++ b/sound/soc/intel/boards/cnl_svfpga.c
@@ -78,11 +78,11 @@ static inline
 struct snd_soc_codec *cnl_svfpga_get_codec(struct snd_soc_card *card)
 {
 	bool found = false;
-	struct snd_soc_component *component;
+	struct snd_soc_codec *codec;
 
-	list_for_each_entry(component, &card->component_dev_list, card_list) {
-		if (!strstr(component->name, "sdw-slave0-14:13:20:05:86:80")) {
-			pr_debug("codec was %s", component->name);
+	list_for_each_entry(codec, &card->codec_dev_list, card_list) {
+		  if (!strstr(codec->component.name, "sdw-slave0-14:13:20:05:86:80")) {
+			pr_debug("codec was %s", codec->component.name);
 			continue;
 		} else {
 			found = true;
@@ -94,7 +94,7 @@ struct snd_soc_codec *cnl_svfpga_get_codec(struct snd_soc_card *card)
 		BUG();
 		return NULL;
 	}
-	return snd_soc_component_to_codec(component);
+	return codec;
 }
 
 static int cnl_svfpga_init(struct snd_soc_pcm_runtime *runtime)
-- 
1.7.5.4

